commit dfa26eda0ddba351687d2e59ebb9171c59d077fd (refs/changes/87/387/2)
Author: David Pacheco <dap@joyent.com>
Date:   2016-09-01T16:41:29+00:00 (3 years, 1 month ago)
    
    MANTA-2974 manta-factoryreset crashes when no "manta" application is present
    Reviewed by: Jordan Paige Hendricks <jordan.hendricks@joyent.com>
    Approved by: Jordan Paige Hendricks <jordan.hendricks@joyent.com>

diff --git a/cmd/manta-factoryreset.js b/cmd/manta-factoryreset.js
index b9aa480..1ef99e2 100755
--- a/cmd/manta-factoryreset.js
+++ b/cmd/manta-factoryreset.js
@@ -192,6 +192,10 @@ async.waterfall([
 		var log = self.log;
 		log.info('checking stage');
 
+		if (!self.application) {
+			return (cb(null));
+		}
+
 		var app = self.application;
 
 		if (app.metadata.SIZE === 'production') {
